### PR TITLE
Decaffeinate CommentBox + improve Talk error messages

### LINKED ISSUE #7021: Improve Talk Error Messages - email verification and banned
## Current behavior
If a user is banned or has an unverified email address (after email verification is deployed -- see https://github.com/zooniverse/talk-api/pull/311), the user is shown an error message when they try to post on Talk: `not allowed to create this Comment`

## Expected / new behavior
Provide an informative error message that describes the reason _why_ the user was unable to post a comment.
- For banned users: "You are banned. Email contact@zooniverse.org for further information."
- For unconfirmed users: "Your account email address is not yet confirmed. Please check your email for confirmation instructions, or visit https://www.zooniverse.org/settings/email to request new confirmation email."
    - Wording based on volunteer feedback in [this Freshdesk ticket](https://zooniverse.freshdesk.com/a/tickets/19169).

### Note Regarding Existing Behavior for DMs:
- Case 1: When trying to send a DM or access a user page (e.g., https://www.zooniverse.org/users/lcjohnso) as a banned user, the error message presented to the user is `You are banned`, matching the response payload from the Talk API (`{"error":"You are banned"}`).
- Case 2: When trying to send a DM as a user with an unconfirmed email, the error message presented to the user is `You must confirm your account`, matching the response payload from the Talk API (`{"error":"You must confirm your account"}`).
- Together: these error messages provide OK feedback, and must result from a slightly different error case than when the user is trying to add a post to a Talk thread. Further improvement is a nice-to-have -- see messages above.

## Steps to replicate
Try to post using an account that is unverified or banned.

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview

Staging branch URL: https://pr-7161.pfe-preview.zooniverse.org
Closes #7021

This PR rewrites the `<CommentBox>` component to address two goals: (A) add more descriptive error messages, and (B) convert the code from CoffeeScript to JavaScript

- the CommentBox component is used in 1. creating Discussions, 2. adding a new Comment to a Discussion, and 3. sending direct messages to other users
- Dev note: The semi-automated conversion from CoffeeScript (via [decaffeinate](https://decaffeinate-project.org/)) was just par for the course, as it was easier than trying to keep the CoffeeScript code working.

Expected behaviour:
- If you're logged in as a normal, standard, typical, run-of-the-mill user, you should be able to...
  - Create a discussion on a project (e.g. https://www.zooniverse.org/projects/darkeshard/test-project-2022/talk/269?env=staging)
  - Create a comment in a discussion (e.g. https://www.zooniverse.org/projects/darkeshard/test-project-2022/talk/269/407?env=staging)
  - Send direct messages to other volunteers (https://www.zooniverse.org/inbox)
- If you're a user that hasn't confirmed their email yet, then...
  - ...you should not be able to do any of the above,
  - 🆕 ...and the **error message** when you click on the **"Submit"** button should be `"Your account email address is not yet confirmed. Please check your email for confirmation instructions, or visit https://www.zooniverse.org/settings/email to request new confirmation email."`
  - (Previously, this error message would either be `"not allowed to create this Discussion"` or `"not allowed to create this Discussion"` or `You must confirm your account `)
- If you're a user that has been banned, then...
  - ...you should not be able to do any of the above,
  - 🆕 ...and the **error message** when you click on the **"Submit"** button should be `You are banned. Email [contact@zooniverse.org](mailto:contact@zooniverse.org) for further information.`
  - (Previously, this error message would just be `"You are banned"`)
  - ⚠️ MAJOR NOTE: a banned user may actually have trouble even reaching the pages which have the components that implement these error messages, but that's beyond the scope of this PR.

### Testing

Testing steps:
- Login with a standard run-of-the-mill user account with a confirmed email address, e.g. `zootester1`
  - Go to https://pr-7161.pfe-preview.zooniverse.org/projects/darkeshard/test-project-2022/talk/269?env=staging and create a discussion.
  - Go to https://pr-7161.pfe-preview.zooniverse.org/projects/darkeshard/test-project-2022/talk/269/407?env=staging and post a comment 
  - Go to https://pr-7161.pfe-preview.zooniverse.org/inbox?env=staging and send a message (to `darkeshard` or `zootester2` or something)
  - Confirm that behaviour matches expected behaviour.
- Login with a user account with an unconfirmed email address and that's _not_ banned, e.g. `zootester2024b`
  - Repeat all of the above
- Login with any test account, e.g. zootester2024b
  - Repeat all of the above, BUT at _each step,_ right before pressing the "Submit" button, ban the user account (via a separate window where you're logged in as an admin)
  - Then, un-ban the test account before moving to the next step.
  - This is required because _a banned user can't even load the Talk boards to begin with._ 

NOTE: being banned supersedes being unconfirmed, so a banned && unconfirmed user will only receive "you are banned" error messages.

### Status

Ready for review.

I'd appreciate some very thorough tests on posting comments, as this affects Talk.

### CODE REVIEW

File: app/talk/comment-box.jsx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      return ERROR_MESSAGES.EMAIL_NOT_CONFIRMED;
+    }
+  } else if (
+    error.message?.match?.(/You are banned/i)
+  ) {
+    return ERROR_MESSAGES.BANNED;
+  }
+
+  return error.message || 'Unknown error';
+}
```
Comment (shaunanoordin) [inline app/talk/comment-box.jsx:37]: Dev Notes 1/2: this is one of two additions/modifications to the CommentBox code.

In these blocks of code, we add a function for converting server-side error messages into something more detailed for the volunteer. Error message text is taken directly from #7021.

The rest of the changes is just a result of decaffeinating the CoffeeScript.

File: app/talk/comment-box.jsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    this.setState({loading: true});
+
+    return this.props.onSubmitComment?.(e, textareaValue, this.state.subject, this.state.reply)
+      .then(() => {
+        this.hideChildren();
+        this.setState({subject: null, content: '', error: '', loading: false});
+        return this.setFeedback(this.props.submitFeedback);
+    }).catch(e => {
+        const errorMessage = getErrorMessage(e, this.props.user);
+        return this.setState({error: errorMessage, loading: false});
```
Comment (shaunanoordin) [inline app/talk/comment-box.jsx:109]: Dev Notes 2/2: this is one of two additions/modifications to the CommentBox code.

Here, we use the new `getErrorMessage()` to determine the appropriate error message for volunteers.

The rest of the changes is just a result of decaffeinating the CoffeeScript.

**Comment by shaunanoordin** [comment]
## PR Update:

- Fix: other CoffeeScript Talk components that import (`require`) the new decaffeinated CoffeeScript actually require `.default`. Not sure why this didn't pop up on my local testing earlier.
- If you try to post messages, and you receive `Request has been terminated Possible causes: the network is offline, Origin is not allowed by Access-Control-Allow-Origin, the page is being unloaded, etc.`, then... the Talk Staging API is probably down. 😓 Sorry staging API! I think my non-stop tests might have been a bit too much for you.

**Review by kieftrav — APPROVED** [review]
PR looks good with no suggested changes! 

- Translation via decaffeinate matches on all objects, functions, and parameters.
- Additional code is minimal, clear, and straightforward
- All user testing scenarios (unconfirmed, confirmed, and banned) tested and worked as expected

This PR is good to go from my perspective!

**Review by lcjohnso — APPROVED** [review]
Looks great! I confirm that testing via branch deploy produces expected result; this is good to go!

**Comment by goplayoutside3** [comment]
I'm going to go ahead and deploy this! Totally okay to make this change in PFE even among the launching new pages effort.

**Comment by goplayoutside3** [comment]
I'm going to go ahead and deploy this! Totally okay to make this change in PFE even among the launching new pages effort.

**Pull request merged**